src/Microsoft.VisualStudio.Services.Agent/Util/VarUtil.cs

```diff
@@ -63,6 +87,28 @@ public static void ExpandEnvironmentVariables(IHostContext context, IDictionary<
             ExpandValues(context, source, target);
         }
 
+        public static JToken ExpandValues(IHostContext context, IDictionary<string, string> source, JToken target)
+        {
+            var mapFuncs = new Dictionary<JTokenType, Func<JToken, JToken>>
+            {
+                {
+                    JTokenType.String,
```

Comment:
null?

Reply:
JTokenType.Null will just return the original value